canary/generator/download/download_split_by.py

```diff
@@ -0,0 +1,68 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from collections import defaultdict
+from multiprocessing import Pool
+from itertools import product
+import json
+from json import JSONDecodeError
+from ast import literal_eval
+import os
+import sys
+
+import requests
+import numpy as np
+
+from .query import build_query_string
+
+
+def _get_data(arglist):
+    dates_for_each, m, ch, v, os, app, ar = arglist
+    q = build_query_string(dates_for_each, m, ch, v, os, application=app, architecture=ar)
+    data = requests.get(q).text
+    try:
+        data = json.loads(data)
+    except JSONDecodeError:
+        print('bad metric: ', m)
+        return
+    data['channel'] = ch
+    data['metric'] = m
+    data['version'] = v
+    data['os'] = os
+    data['application'] = app
+    data['architecture'] = ar
+    return data
+
+
+if __name__ == '__main__':
+    pool = Pool(20)
+    channels = requests.get(
+        'https://aggregates.telemetry.mozilla.org/aggregates_by/build_id/channels/').text
+    channels = literal_eval(channels)
+    ch = 'nightly'
+    dates_versions = requests.get(
+        'https://aggregates.telemetry.mozilla.org/'
```

Comment:
The base URL is used 3 times: I think it's worth factoring it out to a constant and use it